### PR TITLE
add config check that validators track all shards

### PR DESCRIPTION
**Pull request by mzhangmzz**

### CODE REVIEW

File: nearcore/src/config.rs [hunk 1]
```diff
@@ -1225,6 +1227,7 @@ pub fn load_config(
         let signer = InMemoryValidatorSigner::from_file(&validator_file).with_context(|| {
             format!("Failed initializing validator signer from {}", validator_file.display())
         })?;
+        assert!(!config.tracked_shards.is_empty(), "Validator must track all shards. Please change `tracked_shards` field in config.json to be any non-empty vector");
```
Comment (matklad) [inline nearcore/src/config.rs:1230]:
```suggestion
        anyhow::ensure!(!config.tracked_shards.is_empty(), "Validator must track all shards. Please change `tracked_shards` field in config.json to be any non-empty vector")?;
```

We have nice error handling via `?` setup here, no need to assert.

**Comment by matklad** [comment]
I'll let @mm-near to rubber-stump this: I don't really understand how this config works, the logic around "not empty => track everything" always seemed funky to me.

File: nearcore/src/config.rs [hunk 2]
```diff
@@ -825,6 +825,7 @@ pub fn init_configs(
     }
 
     let mut config = Config::default();
+    config.tracked_shards = vec![0];
```
Comment (mm-near) [inline nearcore/src/config.rs:828]: btw - seems that the config is actually downloaded/changed in the next lines, so maybe we should move this later.

File: nearcore/src/config.rs [hunk 3]
```diff
@@ -1063,6 +1064,7 @@ pub fn create_testnet_configs_from_seeds(
             };
             config.network.skip_sync_wait = num_validator_seats == 1;
         }
+        config.tracked_shards = vec![0];
```
Comment (mm-near) [inline nearcore/src/config.rs:1067]: comment (mention https://github.com/near/nearcore/issues/7388)

File: nearcore/src/config.rs (hunk 1 above)
Comment (mm-near) [inline nearcore/src/config.rs:1230]: You can also add link to: https://github.com/near/nearcore/issues/7388 in the message

**Review by mm-near — APPROVED** [review]

**Pull request merged**